### ISSUE TITLE
Config Changes for Craft 3.5 to avoid Deprecation Error

### ISSUE THREAD

**Issue by ghost** [fortrabbit/craft-object-storage#8]
Hi there,

Installing Craft 3.5.8 with Object Storage 1.10 reveals this deprecation error:

Support for overriding volume configs in config/volumes.php has been deprecated. Environment variables or dependency injection should be used instead.

volumes.php looks like this for me:
```
<?php
$objectStorageAccess = [
    'url'      => "https://" . getenv('OBJECT_STORAGE_HOST'),
    'endpoint' => "https://" . getenv('OBJECT_STORAGE_SERVER'),
    'keyId'    => '$OBJECT_STORAGE_KEY',
    'secret'   => '$OBJECT_STORAGE_SECRET',
    'bucket'   => getenv('OBJECT_STORAGE_BUCKET'),
    'region'   => getenv('OBJECT_STORAGE_REGION'),
];

return [
    // Default volume
    'objectStorageAssets' => array_merge($objectStorageAccess, [
        'name'      => 'Site Images',
        'subfolder' => 'site_images',
    ]),
    'objectStorageAssets2' => array_merge($objectStorageAccess, [
        'name'      => 'Banner Images',
        'subfolder' => 'banner_images',
    ]),
    'objectStorageAssets3' => array_merge($objectStorageAccess, [
        'name'      => 'Profile Pictures',
        'subfolder' => 'profile_pictures',
    ]),
];

```

Does the volumes config file need to be changed to be compatible with future versions of Craft?

**Comment by ghost** [comment]
3.5.8 deprecation listed here: https://github.com/craftcms/cms/blob/master/CHANGELOG.md

**Comment by roelvanhintum** [comment]
Pull #9 fixes this.

**Comment by ostark** [comment]
Thanks @mjatharvest for highlighting this and @roelvanhintum for creating the PR 🥳

**Comment by roelvanhintum** [comment]
@ostark, I hope i didn't strip to much functionality. The idea is to use Craft's environment configuration in the volume settings form. https://craftcms.com/docs/3.x/config/#environmental-configuration
As suggested on https://github.com/craftcms/cms/blob/develop/CHANGELOG.md#358---2020-09-01

**Comment by ostark** [comment]
I'm happy to remove code, as long as it does not increase the efforts of configuration.

**Comment by howells** [comment]
@roelvanhintum nice work on the PR! Would there be any chance of the Subfolder field being able to use an env variable or alias? In my scenario I use $ENVIRONMENT to determine the Subfolder

**Comment by howells** [comment]
(I went ahead and created pull #10 to do this.)

**Comment by roelvanhintum** [comment]
@howells, good point. I did the same for my pull request, in case that gets merged.

**Comment by jamiepittock** [comment]
@ostark will the PR that fixes this be merged?

**Comment by ostark** [comment]
I'll continue here https://github.com/fortrabbit/craft-object-storage/tree/feature/craft35-compat

**Comment by ostark** [comment]
Released as 2.0, since there might be tiny breaking changes. 
https://github.com/fortrabbit/craft-object-storage/blob/master/CHANGELOG.md
